### PR TITLE
Align lines with different columns count

### LINKED ISSUE #1783: Align lines with different columns count
This template is a guideline, not a strict requirement.
- **Version**: 2.4.2
- **Configuration**:
```
version = 2.4.2
align = most
```

## Steps

Given code like this:
```scala
libraryDependencies ++= Seq(
"io.get-coursier" % "interface" %% "0.0.17",
"com.geirsson1111111" %%% "metaconfig-core" % metaconfigV % Test,
"org.scalacheck" %% "scalacheck" % sclacheckV % Test
)
```

When I run scalafmt like this:

```bash
scalafmt
```

## Problem

Scalafmt formats code like this:
```scala
libraryDependencies ++= Seq(
  "io.get-coursier"       % "interface"      %% "0.0.17",
  "com.geirsson1111111" %%% "metaconfig-core" % metaconfigV % Test,
  "org.scalacheck"       %% "scalacheck"      % sclacheckV % Test
)
```

## Expectation

I would like the formatted output to look like this:
```scala
libraryDependencies ++= Seq(
  "io.get-coursier"       % "interface"      %% "0.0.17",
  "com.geirsson1111111" %%% "metaconfig-core" % metaconfigV % Test,
  "org.scalacheck"       %% "scalacheck"      % sclacheckV  % Test
)
```

## Notes

Based on https://github.com/scalameta/scalafmt/issues/1534#issuecomment-599141886

### PR DESCRIPTION
**Pull request by poslegm**
Fixes #1783 

TL;DR

```scala
//before
libraryDependencies ++= Seq(
  "a"   % "x" %% "1",
  "b" %%% "y"  % "2.2.2.2" % Test,
  "c"  %% "z"  % "3" % Test
)
//after
libraryDependencies ++= Seq(
  "a"   % "x" %% "1",
  "b" %%% "y"  % "2.2.2.2" % Test,
  "c"  %% "z"  % "3"       % Test
)

### CODE REVIEW

File: scalafmt-core/shared/src/main/scala/org/scalafmt/internal/FormatWriter.scala [hunk 1]
```diff
@@ -663,7 +663,7 @@ class FormatWriter(formatOps: FormatOps) {
     val units = Vector.newBuilder[AlignmentUnit]
     while (i < block.length) {
       val line = block(i)
-      if (line.lengthCompare(column) == 1) {
+      if (column < line.length) {
```
Comment (kitbellew) [inline scalafmt-core/shared/src/main/scala/org/scalafmt/internal/FormatWriter.scala]: also, if you like `lengthCompare`, `line.lengthCompare(column) > 0`, although not necessarily more readable :)

Comment (poslegm) [reply scalafmt-core/shared/src/main/scala/org/scalafmt/internal/FormatWriter.scala]: When I originally wrote `lengthCompare` I overlooked that `line` is `Array`. So `lengthCompare` is redundant optimization here :)

**Comment by poslegm** [comment]
Also when running this changes on scala-repos I discovered idempotency violations on master:

scala-repos diff: https://github.com/poslegm/scala-repos/commit/4554a0af913af10bb9dfdb51b8c5c5d5070ddf9e

broken example:

```
<<< ONLY step 1
private def getTopN[T](s: Iterable[T], n: Int)(
      implicit ord: Ordering[T]): Seq[T]
>>>
private def getTopN[T](s: Iterable[T], n: Int)(
    implicit ord: Ordering[T]	   
): Seq[T]
<<< ONLY step 2
private def getTopN[T](s: Iterable[T], n: Int)(
    implicit ord: Ordering[T]	   
): Seq[T]
>>>
private def getTopN[T](s: Iterable[T], n: Int)(implicit
    ord: Ordering[T]
): Seq[T]
```

(works correct at 2.4.2)

Is it related to your latest changes?

**Comment by kitbellew** [comment]
> @kitbellew Also when running this changes on scala-repos I discovered idempotency violations on master:
> 
> scala-repos diff: [poslegm/scala-repos@4554a0a](https://github.com/poslegm/scala-repos/commit/4554a0af913af10bb9dfdb51b8c5c5d5070ddf9e)
> 
> broken example:
> 
> ```
> <<< ONLY step 1
> private def getTopN[T](s: Iterable[T], n: Int)(
>       implicit ord: Ordering[T]): Seq[T]
> >>>
> private def getTopN[T](s: Iterable[T], n: Int)(
>     implicit ord: Ordering[T]	   
> ): Seq[T]
> <<< ONLY step 2
> private def getTopN[T](s: Iterable[T], n: Int)(
>     implicit ord: Ordering[T]	   
> ): Seq[T]
> >>>
> private def getTopN[T](s: Iterable[T], n: Int)(implicit
>     ord: Ordering[T]
> ): Seq[T]
> ```
> 
> (works correct at 2.4.2)
> 
> Is it related to your latest changes?

probably :( `implicit` handling is new. can you disable them for now, open a ticket for me to track, please?

**Comment by poslegm** [comment]
https://github.com/scalameta/scalafmt/issues/1787
